silkworm/rpc/commands/parity_api.cpp

```diff
@@ -118,23 +132,30 @@ Task<void> ParityRpcApi::handle_parity_list_storage_keys(const nlohmann::json& r
         std::optional<Account> account = co_await state_reader.read_account(address);
         if (!account) throw std::domain_error{"account not found"};
 
-        Bytes seek_bytes = db::storage_prefix(address.bytes, account->incarnation);
-        const auto cursor = co_await tx->cursor_dup_sort(db::table::kPlainStateName);
-        SILK_TRACE << "ParityRpcApi::handle_parity_list_storage_keys cursor id: " << cursor->cursor_id();
-        Bytes seek_val = offset ? offset.value() : Bytes{};
-
-        std::vector<evmc::bytes32> keys;
-        auto v = co_await cursor->seek_both(seek_bytes, seek_val);
-        // We look for keys until we have the quantity we want or the key is invalid/empty
-        while (v.size() >= kHashLength && keys.size() != quantity) {
-            auto value = bytes32_from_hex(silkworm::to_hex(v));
-            keys.push_back(value);
-            const auto kv_pair = co_await cursor->next_dup();
-
-            if (kv_pair.key != seek_bytes) {
-                break;
-            }
-            v = kv_pair.value;
+        const auto txn_number = co_await tx->first_txn_num_in_block(block_number);
+        auto from = db::code_domain_key(address);
+
+        if (offset) {
+            from.append(offset.value());
+        }
+        auto to = db::code_domain_key(address);
+        increment(to);
+        SILK_DEBUG << "handle_parity_list_storage_keys: from " << from << ", to " << to;
+
+        db::kv::api::DomainRangeQuery query{
+            .table = db::table::kStorageDomain,
+            .from_key = from,
+            .to_key = to,
+            .timestamp = txn_number,
+            .ascending_order = true,
+            .limit = quantity};
+        auto paginated_result = co_await tx->domain_range(std::move(query));
+        auto it = co_await paginated_result.begin();
+
+        std::vector<std::string> keys;
+        while (const auto value = co_await it.next()) {
+            const auto key = value->first.substr(20);
```

Comment:
It would be better to be less aggressive here:
- check that length of `value->first` is greater than `kAddressLength`
- use `kAddressLength` instead of hard-coded value